compiler/pragmas.nim

```diff
@@ -586,15 +586,7 @@ proc pragmaEmit(c: PContext, n: PNode) =
 proc noVal(c: PContext; n: PNode) =
   if n.kind in nkPragmaCallKinds and n.len > 1: invalidPragma(c, n)
 
-proc pragmaUnroll(c: PContext, n: PNode) =
-  if c.p.nestedLoopCounter <= 0:
-    invalidPragma(c, n)
-  elif n.kind in nkPragmaCallKinds and n.len == 2:
-    var unrollFactor = expectIntLit(c, n)
-    if unrollFactor <% 32:
-      n[1] = newIntNode(nkIntLit, unrollFactor)
-    else:
-      invalidPragma(c, n)
+proc pragmaUnroll(c: PContext, n: PNode) {.deprecated: "Deprecated since v1.3".} = discard #14695
```

Comment:
That's now how it works. The *compiler* should produce a deprecation warning for when you use ``.unroll``